test/e2e/tls_test.go

```diff
@@ -0,0 +1,517 @@
+package e2e
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/imroc/req/v3"
+	"github.com/stretchr/testify/assert"
+	"github.com/tidwall/gjson"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	http_helper "github.com/gruntwork-io/terratest/modules/http-helper"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/random"
+)
+
+func TestTLSEnabledWithSelfSigned(t *testing.T) {
+	// Path to the helm chart we will test
+	helmChartPath, e := filepath.Abs("../../charts")
+	if e != nil {
+		t.Fatalf(e.Error())
+	}
+	username := "admin"
+	password := "admin"
+
+	namespaceName := "marklogic-" + strings.ToLower(random.UniqueId())
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+	options := &helm.Options{
+		KubectlOptions: kubectlOptions,
+		SetValues: map[string]string{
+			"persistence.enabled":           "false",
+			"replicaCount":                  "1",
+			"image.repository":              "marklogicdb/marklogic-db",
+			"image.tag":                     "latest",
+			"auth.adminUsername":            username,
+			"auth.adminPassword":            password,
+			"logCollection.enabled":         "false",
+			"tls.enableOnDefaultAppServers": "true",
+		},
+	}
+
+	t.Logf("====Creating namespace: " + namespaceName)
+	k8s.CreateNamespace(t, kubectlOptions, namespaceName)
+
+	defer t.Logf("====Deleting namespace: " + namespaceName)
+	defer k8s.DeleteNamespace(t, kubectlOptions, namespaceName)
+
+	t.Logf("====Installing Helm Chart")
+	releaseName := "test-join"
```

Comment:
Release name should be shorten to avoid FQDN 64 characters issue.

Reply:
Tested this FQDN locally with the 64-character limit, and this name works fine.